soccer/gameplay/plays/offense/distraction.py

```diff
@@ -0,0 +1,201 @@
+
+                        (self.has_subbehavior_with_name('distract pass') and 
+                            self.subbehavior_with_name('distract pass').is_done_running()),
+                        'if the ball is already close to the ball, then go from the striker captureing the ball to shooting the ball')
+
+        self.add_transition(distraction.State.cross,
+                        distraction.State.shoot, lambda: 
+                        self.subbehavior_with_name('pass to striker').is_done_running() or 
+
+                        (self.has_subbehavior_with_name('capture 3') and 
+                            self.subbehavior_with_name('capture 3').is_done_running()) , 
+                        'go from the striker receiving the pass to shooting the ball')
+
+        self.add_transition(distraction.State.shoot,
+                        distraction.State.passing, lambda: self.dont_shoot,
+                        "don't shoot if shoot chance is too low")
+
+        self.add_transition(distraction.State.shoot,
+                        distraction.State.setup, lambda: 
+                        self.subbehavior_with_name('shooting').is_done_running(), 
+                        'repeat')
+
+    def on_enter_setup(self):
+        self.remove_all_subbehaviors()
+        #capture ball and get striker and distractor in position
+        self.add_subbehavior(skills.capture.Capture(), 'capture', required = True)
+        
+        if not self.distracter_get_close_ball:
+            self.add_subbehavior(skills.move.Move(self.distraction_recieve_pass_point), 'distract moves', required = True)
+        
+        if not self.striker_get_close_ball:
+            self.add_subbehavior(skills.move.Move(self.striker_point), 'striker moves', required = False, priority = 10) 
+        
+    def on_enter_optional_adjustment(self):
+        self.remove_all_subbehaviors()
+        #if the ball is too far then the distractor moves to the center
+        self.add_subbehavior(skills.capture.Capture(), 'capture 2', required = True)
+        self.add_subbehavior(skills.move.Move(self.center), 'move half', required = True)
+        self.add_subbehavior(skills.move.Move(self.striker_point), 'make striker stay', required = True)
+
+    def on_enter_center_pass(self):
+        self.remove_all_subbehaviors()
+        #pass the ball to the robot in the center and move a robot to the distract position 
+        self.add_subbehavior(tactics.coordinated_pass.CoordinatedPass(self.center), 'center pass', required = True)
+        self.add_subbehavior(skills.move.Move(self.distraction_recieve_pass_point), 'move back to distract', required = False, priority = 10)
+
+    def on_enter_passing(self):
+        self.remove_all_subbehaviors()
+        #either pass to striker or distracter depending on shot chance        
+        pass_to_distract_chance = evaluation.passing.eval_pass( main.ball().pos, self.distraction_recieve_pass_point, main.our_robots() )
```

Comment:
These arguments might need to be on another line. 80-100 chars long is about my limit on what's ok, but I'm not super consistent. It's basically a style choice